rust/src/clib/test/Makefile

```diff
@@ -1,36 +1,42 @@
+
 TMPDIR := $(shell mktemp -d)
 TOP_SRC_DIR:=$(shell dirname $(realpath $(firstword \
 	$(MAKEFILE_LIST))))/../../../
-DEBUG_NMSTATE_SO_FILE=$(TOP_SRC_DIR)/target/debug/libnmstate.so
 
 include $(TOP_SRC_DIR)/Makefile.inc
 
+RUST_DEBUG_BIN_DIR=$(TOP_SRC_DIR)/target/debug
+CLIB_SO_DEV_DEBUG=$(RUST_DEBUG_BIN_DIR)/$(CLIB_SO_DEV)
+CLIB_HEADER=nmstate.h
+CLIB_HEADER_IN=$(TOP_SRC_DIR)src/clib/$(CLIB_HEADER).in
+
 CFLAGS =-g -Wall -Wextra -L$(TMPDIR)
 LDFLAGS =-I$(TMPDIR)
 LIBS =-lnmstate
 CC ?=cc
 
 all: check
 
-HEADER_IN = $(TOP_SRC_DIR)/src/clib/nmstate.h.in
-
-nmstate.h: $(HEADER_IN)
-	install -p -v -D -m644 $(HEADER_IN) \
-		$(TMPDIR)/nmstate.h
+nmstate.h: $(CLIB_HEADER_IN)
+	install -p -v -D -m644 $(CLIB_HEADER_IN) \
+		$(TMPDIR)/$(CLIB_HEADER)
 	sed -i -e 's/@_VERSION_MAJOR@/$(CLIB_VERSION_MAJOR)/' \
-		$(TMPDIR)/nmstate.h
+		$(TMPDIR)/$(CLIB_HEADER)
 	sed -i -e 's/@_VERSION_MINOR@/$(CLIB_VERSION_MINOR)/' \
-		$(TMPDIR)/nmstate.h
+		$(TMPDIR)/$(CLIB_HEADER)
 	sed -i -e 's/@_VERSION_MICRO@/$(CLIB_VERSION_MICRO)/' \
-		$(TMPDIR)/nmstate.h
+		$(TMPDIR)/$(CLIB_HEADER)
+
+$(CLIB_SO_DEV_DEBUG):
+	cd $(TOP_SRC_DIR) && cargo build -p nmstate-clib
 
-libnmstate.so: $(DEBUG_NMSTATE_SO_FILE)
-	install -p -v -D -m755 $(DEBUG_NMSTATE_SO_FILE) \
+install: $(CLIB_SO_DEV_DEBUG)
+	install -p -v -D -m755 $(CLIB_SO_DEV_DEBUG) \
 		$(TMPDIR)/$(CLIB_SO_FULL)
 	ln -sfv $(CLIB_SO_FULL) $(TMPDIR)/$(CLIB_SO_MAN)
 	ln -sfv $(CLIB_SO_FULL) $(TMPDIR)/$(CLIB_SO_DEV)
 
-nmstate_test: nmstate_test.c nmstate.h libnmstate.so
+nmstate_test: nmstate_test.c nmstate.h install
```

Comment:
Please use `CGO_LDFLAGS` and `CGO_CFLAGS` instead of install it. I ran `make check` in my OS which I don't want it been installed in this case

Reply:
I will copy (although it introduces some redundancy) the build/install code the the src/go/nmstate/Makefile